### PR TITLE
vendor in 1.23.5 CDI

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

Main reason for this is the updated clone authorization function.  Is is needed to handle the "golden images" namespace cloning.  The issue was that serviceaccount groups were not considered for auth decisions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by awels — COMMENTED** [review]
/lgtm

File: vendor/kubevirt.io/containerized-data-importer/pkg/clone/auth.go [hunk 1]
```diff
@@ -55,21 +59,22 @@ func CanUserClonePVC(client kubernetes.Interface, sourceNamespace, pvcName, targ
...
 	if pvcNamespace == saNamespace {
 		return true, "", nil
 	}
 
 	user := fmt.Sprintf("system:serviceaccount:%s:%s", saNamespace, saName)
 
 	sarSpec := authorization.SubjectAccessReviewSpec{
-		User: user,
+		User:   user,
+		Groups: []string{"system:serviceaccounts", "system:serviceaccounts:" + saNamespace},
```
Comment (mhenriks) [inline vendor/kubevirt.io/containerized-data-importer/pkg/clone/auth.go:71]: This is the relevant bit here.

**Pull request merged**